tests/test_updater_validation.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+# Copyright 2022, New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""Test ngclient Updater validations.
+"""
+
+import os
+import sys
+import tempfile
+import unittest
+
+import securesystemslib.exceptions
+
+from tests import utils
+from tests.repository_simulator import RepositorySimulator
+from tuf import unittest_toolbox
+from tuf.api.metadata import TargetFile
+from tuf.exceptions import FetcherHTTPError
+from tuf.ngclient import Updater
+
+
+class TestUpdater(unittest_toolbox.Modified_TestCase):
+    """Test ngclient Updater input validation."""
+
+    def setUp(self) -> None:
+        # pylint: disable-next=consider-using-with
+        self.temp_dir = tempfile.TemporaryDirectory()
+        self.metadata_dir = os.path.join(self.temp_dir.name, "metadata")
+        self.targets_dir = os.path.join(self.temp_dir.name, "targets")
+        os.mkdir(self.metadata_dir)
+        os.mkdir(self.targets_dir)
+
+        # Setup the repository, bootstrap client root.json
+        self.sim = RepositorySimulator()
+        with open(os.path.join(self.metadata_dir, "root.json"), "bw") as f:
+            f.write(self.sim.signed_roots[0])
+
+    def tearDown(self) -> None:
+        self.temp_dir.cleanup()
+
+    def _new_updater_with_targets_setup(self) -> Updater:
+        return Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            self.targets_dir,
+            "https://example.com/targets/",
+            fetcher=self.sim,
+        )
+
+    def _new_updater_without_targets_setup(self) -> Updater:
+        return Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            fetcher=self.sim,
+        )
+
+    def _new_updater_with_non_existing_metadata_dir(self) -> Updater:
+        return Updater(
+            "non_existing_metadata_dir",
+            "https://example.com/metadata/",
+            fetcher=self.sim,
+        )
+
```

Comment:
Those are just for better readability, because I reuse `_new_updater_with_targets_setup` multiple times and if it's just `_new_updater` it wouldn't be clear why it's not used everywhere, while now the naming is informative and the same approach is used everywhere.

Reply:
I think because they are functions it adds overhead.
Maybe add a comment instead.